edb/server/protocol/auth_ext/util.py

```diff
@@ -101,28 +103,38 @@ def get_config_typename(config_value: edb_config.SettingValue) -> str:
     return config_value._tspec.name  # type: ignore
 
 
+def escape_and_truncate(input_str: str | None, max_len: int) -> str | None:
+    if input_str is None:
+        return None
+    return html.escape(input_str)[:max_len]
```

Comment:
escape should happen on the trimmed string, not after -- can risk having broken escape codes